### PR TITLE
Add an app to update the email signature PNG daily.

### PR DESCRIPTION
**Pull request by YPCrumble**
@lwm in the spirit of moving quickly, this is a preliminary PR for adding the email signature png.

The path should be:

    https://api.carbondoomsday.com/media/carbondoomsday_email_signature.png

I wrote a couple tests for the Python portion of the implementation, but I'm less confident about my celery and dokku integrations...and not sure how to build out the staging server so they're untested. Would love any feedback and happy to make adjustments as required. Thanks!

### CODE REVIEW

**Comment by decentral1se** [comment]
I'll get to a proper review on this later today or tomorrow!

Thanks for coming back to this :)

EDIT: CI seems to be complaining about missing a `pipenv lock` or whatever.

File: carbondoomsday/settings.py [hunk 1]
```diff
@@ -141,6 +141,13 @@ class CeleryAndCeleryBeat():
             ),
             'schedule': timedelta(minutes=50)
         },
+        'update-email-signature': {
+            'task': (
+                'carbondoomsday.email_signature.tasks.'
+                'update_email_signature'
+            ),
+            'schedule': timedelta(days=1)
```
Comment (decentral1se) [inline carbondoomsday/settings.py]: Perhaps this should be fired manually from the other celery tasks. For example, if the latest data updates at some strange time and this runs before that, the signature will not have the latest data. I think you'll want to try with `apply_async`.

File: carbondoomsday/email_signature/utils.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Inspiration from:
+# https://code-maven.com/create-images-with-python-pil-pillow
+# https://stackoverflow.com/a/38629258/2532070
+# https://stackoverflow.com/a/273962/2532070
+# https://stackoverflow.com/a/2563883/2532070
+# add_border_radius method:
+# https://stackoverflow.com/a/11291419/2532070
```
Comment (decentral1se) [inline carbondoomsday/email_signature/utils.py]: If these are useful to help understand the following code then I'd rather see them closer to the code that is using something from them. That might be in a function docstring in the usual `""" my docstring """`? If more general, could also use the module docstring at the top here.

**Comment by decentral1se** [comment]
Looks good! Nice one! Although I do question whether we need to the separate app for it? I imagined it being included in the original carbondoomsday app.

I'm going to submit a PR (see https://github.com/giving-a-fuck-about-climate-change/carbondoomsday/pull/123) to clear up our current CI pipenv issues. That might give you some conflict issues which you'll need to rebase over and resolve but you'll get a clean CI run then!

**Comment by YPCrumble** [comment]
@lwm do you mean to day the "measurements" app? I didn't think the signature was really a measurement so put it in a new app. Happy to move it to the measurements app if you prefer. I usually try to make apps do one thing only and most of my apps are quite simple, but I really don't know if that's a good practice or not!

**Comment by lwm** [comment]
>  usually try to make apps do one thing only

**Comment by decentral1se** [comment]
>  usually try to make apps do one thing only

I approve of your zealous UNIX attitude! Let's do it.

**Comment by decentral1se** [comment]
The CI failures probably needs a similar to https://github.com/giving-a-fuck-about-climate-change/carbondoomsday/pull/121#issuecomment-390923225 treatment?

File: carbondoomsday/email_signature/utils.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+import datetime
+import sys
+from PIL import Image, ImageFont, ImageDraw, ImageEnhance
+
+from carbondoomsday.measurements.models import CO2
+
+
+# ppm = "412.32"
```
Comment (decentral1se) [inline carbondoomsday/email_signature/utils.py]: We can remove this?

**Comment by decentral1se** [comment]
OK, let's get this PR going green and I will try to do some background work to get you able to make deployments and such. Then we deploy and therefore win.

**Comment by YPCrumble** [comment]
@lwm I think I'm getting closer - the isort functionality is still confusing me.

This is the result from TravisCI: https://cl.ly/0e0y3K3I142M

This is the result when I run `make sort` on my machine after making the change suggested by TravisCI: https://cl.ly/0z1W1N1d2v1f

**Comment by YPCrumble** [comment]
@lwm nevermind - this is related to the fact that `PIL` is actually `Pillow` and making isort fritz out. Adding it to `known_third_party` in `setup.cfg` should fix; relaunching TravisCI now.

**Comment by YPCrumble** [comment]
@lwm I'm getting closer! You run a tight ship here. Codecov is complaining because my settings.py file isn't fully tested. My suggestion is we either reduce the coverage % or exclude settings.py from coverage - does one or the other (or something else) work for you?

**Comment by decentral1se** [comment]
Nice one!

> You run a tight ship here. 

:laughing: 

TBH, I'm easy with whatever you can configure. The changes looks good to me, from another brief look. Can you send me your SSH public key part? I'll get you access to the machines.

**Comment by YPCrumble** [comment]
Ahhhhh I'm green!

If this now looks good, would you pull in and I can:

1. Check to confirm that the image is getting created and served appropriately
1. Add the email signature to the frontend site!

What's your email to send the public key to @lwm ?

**Comment by decentral1se** [comment]
Send it over to lukewm AT riseup.net (here's [my key](https://pgp.key-server.io/pks/lookup?op=get&search=0x5E2EF5A63E3718CC) if you're into that stuff). I'm going to merge this because I want to get off Github tonight (if possible). I'll add an issue on the migrated repository with those 1. and 2. points!

Thanks for this work! Solid stuff :fire: :fire: :fire:

**Comment by decentral1se** [comment]
:rocket:

**Pull request merged**